### PR TITLE
Chooser dialog refactor

### PR DESCRIPTION
**Pull request by ivolarevic**
# Description

Refactored chooser dialog. In the current implementation, we can only send a certain number of actions. With the new implementation, we can send through the list of actions and methods an unlimited number of parameters for displaying the chooser dialog.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Dev testing.

**Test Configuration**:

* Firmware version: G930FXXS7ETA1
* Hardware: Samsung galaxy s7
* SDK: Android 8, Oreo

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by MatkoTom — CHANGES_REQUESTED** [review]
Please review the comments and make adjustments where necessary.

File: app/src/main/java/com/clover/studio/spikamessenger/utils/extendables/BaseDialog.kt [hunk 1]
```diff
@@ -11,4 +11,5 @@ open class BaseDialog(context: Context?, themeResId: Int?) : AlertDialog(context
 interface DialogInteraction {
     fun onFirstOptionClicked() {}
```
[the remark this answers is not in the document]
Comment (ivolarevic) [reply app/src/main/java/com/clover/studio/spikamessenger/utils/extendables/BaseDialog.kt:12]: We are using these for Error Dialog.

File: app/src/main/java/com/clover/studio/spikamessenger/ui/main/chat_details/ChatDetailsFragment.kt [hunk 2]
```diff
@@ -505,44 +511,59 @@ class ChatDetailsFragment : BaseFragment(), ServiceConnection {
         }
 
         user.formattedDisplayName.let {
```
[the remark this answers is not in the document]
Comment (ivolarevic) [reply app/src/main/java/com/clover/studio/spikamessenger/ui/main/chat_details/ChatDetailsFragment.kt:515]: Done.

**Review by MatkoTom — APPROVED** [review]
LGTM

**Pull request merged**